### DISCUSSION TITLE
support svg

### DISCUSSION THREAD

**Discussion by weiminghuaa** [ebarellar/LFColoringBook#4] [General]
I think support svg will take great experience, because vector graphics are not afraid of scaling. I found that if I can get the clip path, I can use PKCanvasView to draw, and it is very userful with ApplePencil.

```
let mask = CAShapeLayer()
mask.path = cgPath
canvasView.layer.mask = mask
```

But I am stuck on how to get the clip path from the svg, any ideas?

**Comment by ebarellar** [comment]
I’ve also thought about using svg in an updated version 

Instead of using the cgimage from the ColoringImage, an svg image might be used for the layer containing the line drawing. Still, the “coloring” part needs a cgcontext backup in pixels because the Flood Fill approach doesn’t work with vector graphics. To the end users, I doubt they would notice any difference

I haven’t touched PencilKit since the last WWDC, from what I’m aware that type of manipulation is still not supported. Let’s hope we hear good news next month. If they do I’m willing to update the project and replace the custom engine with PencilKit
